src/js/components/Calendar/Calendar.js

```diff
@@ -279,6 +277,22 @@ const Calendar = forwardRef(
       return undefined;
     }, [animating, targetDisplayBounds]);
 
+    useEffect(() => {
+      // Intended for when date changes before the Timeout of a transition ends.
+      // It triggers with changes to `animating` indicating a transition, but it
+      // acts only when `animating` is false, indicating the end of transition.
+      // It verifies the current displayBounds based on date `reference`. If
+      // it doesn't correspond to the wanted bonds, it updates them.
+
+      if (animating === false) {
+        const nextDisplayBounds = buildDisplayBounds(reference, firstDayOfWeek);
+
+        if (nextDisplayBounds !== displayBounds) {
```

Comment:
Won't this always be true, since `buildDisplayBounds()` returns a new array?

Reply:
As mentioned in your last comment, I'll also adapt this to compare each bound as time separately.
The goal is to compare both bounds. If they are not equal, meaning our bounds do not correspond to the current reference, a displayBounds update should be forced.
Comparing arrays is indeed always returning the same thing. I appreciate your help with that :tada: